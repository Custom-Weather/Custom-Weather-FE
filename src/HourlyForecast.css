.hourly-forecast{
  display: inline-block;
  padding: 1px;
  font-size: 12px;
  border-color: white;
}

.icon {
  size: 20%
}
