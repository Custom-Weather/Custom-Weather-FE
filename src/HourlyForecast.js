import React from 'react';
import './HourlyForecast.css';

const HourlyForecast = ({ hour, icon, temp, desc}) => {
  return (
    <div className='hourly-forecast'>
      <h4> Hourly Forecast </h4>
      <img src={`http://openweathermap.org/img/wn/${icon}@2x.png`}></img>
      <p>{hour}</p>
      <p>Current Temp: {temp}</p>
      <p>Description: {desc}</p>
    </div>
  );
};

export default HourlyForecast;
