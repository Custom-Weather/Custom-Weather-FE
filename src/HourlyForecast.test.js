import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';
import HourlyForecast from './HourlyForecast';

describe("HourlyForecast", () => {
  let mockRender;
  beforeEach(() => {
    mockRender=render(
      <HourlyForecast
      hour={"8:00 PM"}
      temp={"16.75"}
      desc={"moderate rain"}
      />
    )
  })

  afterEach(() => {
    cleanup()
  })

  it("should render hourly forecasts", () => {
    const { getByText } = mockRender;
    const hourEl = getByText("8:00 PM")
    expect(hourEl).toBeInTheDocument();
    const tempEl = getByText("16.75")
    expect(tempEl).toBeInTheDocument();
    const descEl = getByText("moderate rain")
    expect(descEl).toBeInTheDocument();
  })
})
