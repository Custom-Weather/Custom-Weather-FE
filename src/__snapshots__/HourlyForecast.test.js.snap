// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="hourly-forecast"
  >
    <p />
    <img
      id="icon"
      src="http://openweathermap.org/img/wn/undefined@2x.png"
    />
    <p />
    <p />
  </div>
</DocumentFragment>
`;
